doc/template.html

```diff
@@ -54,4 +67,24 @@ <h2>Table of Contents</h2>
     </div>
   </div>
 </body>
+<script>
+    'use strict';
+    {
+      if(window.matchMedia) {
+        const mq = window.matchMedia("(prefers-color-scheme: dark)");
+        'onchange' in mq && mq.addEventListener("change", function(e){
+          document.body.classList.toggle("dark-mode", e.matches);
+        })
+        if(mq.matches) {
+          document.body.classList.add("dark-mode");
+        }
+      }
+      const themeToggleButton = document.getElementById("theme-toggle-btn");
+      if (themeToggleButton) {
+        themeToggleButton.addEventListener("click", () => {
+          document.body.classList.toggle("dark-mode");
+        });
+      }
```

Comment:
* Using single quotes instead of double quotes to match the code style of the rest of the code base.
* Use `sessionStorage` to remember preferences across pages
* Disable media query when the user forces a theme
* Remove use of arrow function so the code can be parsed by older browsers.

```suggestion
      const kCustomPreference = 'customDarkTheme';
      const userSettings = sessionStorage.getItem(kCustomPreference);
      const themeToggleButton = document.getElementById('theme-toggle-btn');
      if (userSettings === null && window.matchMedia) {
        const mq = window.matchMedia('(prefers-color-scheme: dark)');
        if ('onchange' in mq) {
          function mqChangeListener(e) {
            document.body.classList.toggle('dark-mode', e.matches);
          }
          mq.addEventListener('change', mqChangeListener);
          if (themeToggleButton) {
            themeToggleButton.addEventListener('click', function() {
              mq.removeEventListener('change', mqChangeListener);
            }, { once: true });
          }
        }
        if (mq.matches) {
          document.body.classList.add('dark-mode');
        }
      } else if (userSettings === 'true') {
        document.body.classList.add('dark-mode');
      }
      if (themeToggleButton) {
        themeToggleButton.hidden = false;
        themeToggleButton.addEventListener('click', function() {
          sessionStorage.setItem(
            kCustomPreference,
            document.body.classList.toggle('dark-mode')
          );
        });
      }
```